src/main/kotlin/com/sourcegraph/cody/agent/protocol/ProtocolTextDocument.kt

```diff
@@ -69,7 +77,8 @@ private constructor(
     @JvmStatic
     fun fromEditorWithRangeSelection(editor: Editor): ProtocolTextDocument? {
       val file = FileDocumentManager.getInstance().getFile(editor.document) ?: return null
-      return ProtocolTextDocument(uri = uriFor(file), selection = getSelection(editor))
+      // Keep current selection as is when sending the document to the agent.
+      return ProtocolTextDocument(uri = uriFor(file), selection = getSelection(editor, false))
```

Comment:
It's often clearer to use named arguments for boolean literals

```suggestion
      return ProtocolTextDocument(uri = uriFor(file), selection = getSelection(editor, isZeroBased = false))
```